sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -5144,6 +5144,17 @@ pi_result cuda_piextUSMEnqueueMemAdvise(pi_queue queue, const void *ptr,
                           PI_MEM_ADVICE_CUDA_SET_PREFERRED_LOCATION)),
           CU_DEVICE_CPU));
       break;
+    case PI_MEM_ADVISE_RESET:
+      PI_CHECK_ERROR(cuMemAdvise((CUdeviceptr)ptr, length,
+                                 CU_MEM_ADVISE_UNSET_READ_MOSTLY,
+                                 queue->get_context()->get_device()->get()));
+      PI_CHECK_ERROR(cuMemAdvise((CUdeviceptr)ptr, length,
+                                 CU_MEM_ADVISE_UNSET_PREFERRED_LOCATION,
+                                 queue->get_context()->get_device()->get()));
+      PI_CHECK_ERROR(cuMemAdvise((CUdeviceptr)ptr, length,
+                                 CU_MEM_ADVISE_UNSET_ACCESSED_BY,
+                                 queue->get_context()->get_device()->get()));
```

Comment:
These two shouldn't be called unless the device has `CU_DEVICE_ATTRIBUTE_CONCURRENT_MANAGED_ACCESS`.

Reply:
I have updated this to check for `CU_DEVICE_ATTRIBUTE_CONCURRENT_MANAGED_ACCESS` for `PI_MEM_ADVICE_RESET`